TODO.md

```diff
@@ -0,0 +1,91 @@
+### Performance notes:
+
```

Comment:
Probably want to remove the things done and turn the rest into issues? historically speaking TODO files rarely get updated.

Reply:
done, created issues / merged into readme